app/plugin.go

```diff
@@ -91,24 +93,47 @@ func (a *App) ActivatePlugins() {
 		active := a.PluginEnv.IsPluginActive(id)
 
 		if pluginState.Enable && !active {
-			if err := a.PluginEnv.ActivatePlugin(id); err != nil {
+			if err := a.activatePlugin(plugin.Manifest); err != nil {
+				failedPluginsToDisable = append(failedPluginsToDisable, id)
 				l4g.Error(err.Error())
 				continue
 			}
 
-			if plugin.Manifest.HasClient() {
-				message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_PLUGIN_ACTIVATED, "", "", "", nil)
-				message.Add("manifest", plugin.Manifest.ClientManifest())
-				a.Publish(message)
-			}
-
-			l4g.Info("Activated %v plugin", id)
 		} else if !pluginState.Enable && active {
 			if err := a.deactivatePlugin(plugin.Manifest); err != nil {
 				l4g.Error(err.Error())
 			}
 		}
 	}
+
+	if len(failedPluginsToDisable) == 0 {
+		return
+	}
+
+	a.UpdateConfig(func(cfg *model.Config) {
+		for _, id := range failedPluginsToDisable {
+			cfg.PluginSettings.PluginStates[id] = &model.PluginState{Enable: false}
+		}
+	})
+
+	if err := a.SaveConfig(a.Config(), true); err != nil {
+		l4g.Error(err.Error())
+	}
```

Comment:
The original problem in the ticket is fixed here. The problem case was if a plugin was set to "Enabled" in config.json, but failed on it's activation, the REST API would report the plugin as deactivated (thus the system console would show it as deactivated) because it did not succeed in activating. Then every time the config changed, the plugin would try to re-activate because the config.json had it as "Enabled". So there was a mismatch in what the API was reporting and what the config.json had

What this does is update the config to match the true state of the plugin if the plugin fails to activate. I'm not entirely convinced this is the best way to go about it so I'm open to opinions

Reply:
Alright yeah, I think that's better too